### PR TITLE
Button to reply to any outstanding transactions

### PR DESCRIPTION
**Pull request by MatthijsKok**

### CODE REVIEW

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/MainActivity.java [hunk 1]
```diff
@@ -45,12 +54,27 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        BunqTransactionParser bunqTransactionParser = new BunqTransactionParser();
+
+        bunqTransactionParser.respondToPendingChallenges(privateKey, transactions, lastSynchronized);
+        lastSynchronized = System.currentTimeMillis();
+
+        showLongToast("Synchronize successful!");
+    }
+
+    private void showLongToast(String text) {
+        Toast.makeText(getApplicationContext(), text, Toast.LENGTH_LONG).show();
```
Comment (RvdK) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/MainActivity.java:77]: This looks liek debug code what does it do

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqTransactionParser.java [hunk 2]
```diff
@@ -23,49 +23,39 @@
...
+        Blockchain chain = teller.getBlockchain();
+
+        transactions.stream()
+                .filter(t -> t.getValue() > 0)
+                .filter(t -> t.getDate().getTime() > cutoffTime)
+                .filter(t -> isValidDescriptionFormat(t.getDescription()))
+                .filter(t -> isValidChallenge(t.getDescription(), chain.getPublicKeyForIBAN(t.getCounterAccount().getIban())))
+                .forEach(t -> {
+                    try {
+                        teller.sendCent(t.getCounterAccount().getIban(),
```
Comment (RvdK) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqTransactionParser.java:37]: this looks dirty to me if this is needed we should probably move the sendcent to the bank interface so we done't have to go back

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/MainActivity.java (hunk 1 above)
Comment (MatthijsKok) [reply BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/MainActivity.java:77]: It show the user that the button action was successful.

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqTransactionParser.java (hunk 2 above)
Comment (MatthijsKok) [reply BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqTransactionParser.java:37]: How is this dirty? I thought that the teller holds a bank object so you dont have to do the difficult setup yourself.

**Review by RvdK — APPROVED** [review]

**Pull request merged**